src/DataSourceProvider.vue

```diff
@@ -154,23 +151,55 @@ export default {
 
       this.loadDataSources(this.widgetData.appId);
     },
+    enableRequiredRules() {
+      this.selectedDataSource.accessRules.forEach(dataSourceRule => {
+        if (dataSourceRule.enabled) {
+          return;
+        }
+
+        // If this rule has any missing access rule
+        this.missingAccessTypes.forEach((missingRule, index) => {
+          if (dataSourceRule.type.includes(missingRule)) {
+            // If this rule for all or for current app
+            if (
+              (!dataSourceRule.appId || dataSourceRule.appId.includes(this.widgetData.appId))
+            ) {
+              // Remove missing access types because we enabled rule where we have it
+              this.missingAccessTypes.splice(index, 1);
+
+              // Enable access rule
+              dataSourceRule.enabled = true;
+            }
+          }
+        });
+      });
+    },
     onAddDefaultSecurity() {
       this.isLoading = true;
       const defaultRules = _.cloneDeep(this.widgetData.accessRules);
 
       if (this.selectedDataSource.accessRules && this.selectedDataSource.accessRules.length > 0) {
+        this.enableRequiredRules();
+
         defaultRules.forEach(defaultRule => {
           defaultRule.type = this.missingAccessTypes;
           defaultRule.enabled = true;
 
-
           let accessRuleFound = this.selectedDataSource.accessRules.some(rule => {
             // Rule considered as duplicated in case if we have the same rule types and same allow option.
-            return defaultRule.allow === rule.allow && !_.difference(rule.type, defaultRule.type).length;
+            // And it's enabled
+            // And it's related to all apps or to the current app
+            return (
+              defaultRule.allow === rule.allow &&
+              !_.difference(rule.type, defaultRule.type).length &&
```

Comment:
Logic operators for multiline expressions should lead with the logic operator and not have them in the end.

```suggestion
              && !_.difference(rule.type, defaultRule.type).length
```

Reply:
@tonytlwu Fixed